### PR TITLE
Create round robin recapture transactions test OT-326

### PR DESCRIPTION
**Pull request by adamtkaczyk**

### CODE REVIEW

File: tests/blockchain/regtest/RoundRobinRecaptureTransactions.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            blocks_number * transaction_in_block_ * amount_in_transaction_;
+    }
+    MineBlocksForUsers(users, target_height, blocks_number);
+
+    // mine MaturationInterval number block with
+    MineBlocks(target_height, static_cast<int>(MaturationInterval()) + 1);
+    target_height += static_cast<int>(MaturationInterval()) + 1;
+
+    for (auto& auxiliary_wallet : auxiliary_wallets_) {
+        auto& user = users_.find(auxiliary_wallet.second.name_)->second;
```
Comment (Piotrek321) [inline tests/blockchain/regtest/RoundRobinRecaptureTransactions.cpp]: if you are sure entry exists (it should - line 54) maybe you can use .at?

File: tests/blockchain/regtest/RoundRobinRecaptureTransactions.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        ot::LogConsole()(user_alice.name_)(" balance: ")(
+            GetDisplayBalance(user_alice))
+            .Flush();
+        ot::LogConsole()(user.name_)(" balance: ")(
+            GetDisplayBalance(sender_balance))
+            .Flush();
+    }
+
+    auto expected_balance = user_alice.expected_balance_;
```
Comment (Piotrek321) [inline tests/blockchain/regtest/RoundRobinRecaptureTransactions.cpp]: const auto &

File: tests/blockchain/regtest/RoundRobinRecaptureTransactions.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    EXPECT_TRUE(Start());
+    EXPECT_TRUE(Connect());
+
+    const auto blocks_number = 2;
+    auto coin_to_send =
+        blocks_number * transaction_in_block_ * amount_in_transaction_ -
+        1000000;
+    auto instance = 2;
+    {
+        auto [user_core, success_core] = CreateClient(
```
Comment (Piotrek321) [inline tests/blockchain/regtest/RoundRobinRecaptureTransactions.cpp]: hmm honestly, I do not see much difference between core and auxiliary wallets in this test. Can you remind me what was the difference?

Comment (adamtkaczyk) [reply tests/blockchain/regtest/RoundRobinRecaptureTransactions.cpp]: Generally division between core and auxiliary is not only in this test but i all Regtest_fixture_simple test as is write in OT-317. In this test for all auxiliary mine coin and send to core wallet. Core wallet have only this coin which received from other wallets (not from mining).

**Review by Piotrek321 — APPROVED** [review]

**Pull request merged**